wordpress-mobile/WordPress-Android
# RI: Fetch "Interests" tags from ReaderTagRepository

This PR 
- Updates `ReaderTagRepository` to return Interest API results to the Interests VM utilising `ReaderRepositoryCommunication` and `FetchInterestTagsUseCase`
- Handles error

~~Looking for an early feedback before moving ahead, so marking it as draft.~~

### To Test

Prerequisite - RI FF flag s enabled

- Launch and go to Reader tab
- Interests from server are shown 

----

- Enable Airplane mode
- Launch and go to Reader tab
- Error layout is displayed with Retry option
- Disable Airplane mode
- Click Retry
- Error layout is hidden and interests are shown

To test remote request failure error, comment out `SuccessWithData` response and return `RemoteRequestFailure` [here](https://github.com/wordpress-mobile/WordPress-Android/blob/f16419db09700f8e9a8985c3944c3a0a662107d7/WordPress/src/main/java/org/wordpress/android/ui/reader/repository/usecases/FetchInterestTagsUseCase.kt#L52).

----

- While the interests are shown, go to My Site -> Me -> App Settings and change interface language (e.g. Spanish)
- Return to Reader tab
- Notice that interests are shown in the selected language (returned from the server).


### ToDo
- Clicking "Comics" interest is not highlighting it 🤷‍♀️, although it enables Done button. Looking into it. 
UPDATE: There are duplicate tags with "Comics" title in the feed. There shouldn't be such a case in the final feed. I'll put a "unique" check to fix it. - 2843b0f
- ~~I'm looking for a better solution for language change support. I've removed `isStarted` check so that we can re-request interests from the server for the changed language but it now happens every time we toggle bottom tabs or rotate screen. I'm thinking to compare last and new language in the interests VM for this something like conditional re-request of tags [here](https://github.com/wordpress-mobile/WordPress-Android/blob/f16419db09700f8e9a8985c3944c3a0a662107d7/WordPress/src/main/java/org/wordpress/android/ui/reader/viewmodels/ReaderViewModel.kt#L76).~~ - 7ac80c2
- Do we want to show "Contact Support" on remote request failure? We haven't yet decided on the error layout design.
- Tests for `FetchInterestTagsUseCase` not yet added. Will include along with other use cases to save interests and get user tags.

---

![device-2020-07-21-190416](https://user-images.githubusercontent.com/1405144/88061490-2d573400-cb85-11ea-9f38-d6477784f464.png)

PR submission checklist:
